sql-plugin/src/main/330db/scala/com/nvidia/spark/rapids/shims/SparkShims.scala

```diff
@@ -52,6 +54,15 @@ object SparkShimImpl extends Spark321PlusDBShims {
     super.getExecs ++ PythonMapInArrowExecShims.execs
 
   override def reproduceEmptyStringBug: Boolean = false
+
+  override def parentReadsShuffleData(shufflePlan: SparkPlan, parent: SparkPlan): Boolean = {
+    val shuffle = shufflePlan.asInstanceOf[ShuffleExchangeLike]
+    parent match {
+      case bhj: GpuBroadcastHashJoinExec =>
+        shuffle.shuffleOrigin.equals(EXECUTOR_BROADCAST)
+      case _ => false
+    }
+  }
```

Comment:
Hmm, I must have somehow dropped this comment from my last review.

The name is a bit misleading since it takes a SparkPlan rather than something more specific, so one could call it with a node that has a parent of ShuffleCoalesceExec and expect this to return true.

I'd rather see this named something like `shuffleParentReadsShuffleData` and have the `shufflePlan` argument be a `ShuffleExchangeLike` for type safety, since that trait should exist on all Spark versions we need.